proxy/src/main/java/com/velocitypowered/proxy/connection/client/ConnectedPlayer.java

```diff
@@ -405,6 +417,7 @@ public ConnectionRequestBuilder createConnectionRequest(RegisteredServer server)
   @Override
   public void setGameProfileProperties(List<GameProfile.Property> properties) {
     this.profile = profile.withProperties(Preconditions.checkNotNull(properties));
+    this.displayName = Component.text(profile.getName());
```

Comment:
Why are you setting the display name back to the original profile name here? By resetting the display name in a place where the name is not concerned, you're creating some minorly surprising behavior.

Reply:
I thought that setting the gameprofile here could change the username, in which case I thought it might make sense to reset the display name, but I'm not sure that's the case